src/poetry/installation/executor.py

```diff
@@ -603,34 +604,66 @@ def _prepare_archive(self, operation: Install | Update) -> Path:
 
         self._populate_hashes_dict(archive, package)
 
-        return self._chef.prepare(archive, editable=package.develop)
+        return self._chef.prepare(
+            archive, editable=package.develop, output_dir=output_dir
+        )
 
     def _prepare_git_archive(self, operation: Install | Update) -> Path:
         from poetry.vcs.git import Git
 
         package = operation.package
+        reference = package.source_resolved_reference or package.source_reference
+        assert package.source_url is not None
+        assert reference is not None
+
+        if package.source_resolved_reference is not None:
```

Comment:
```suggestion
        if package.source_resolved_reference:
```

just in case it's an empty string